packages/webview_flutter/test/webview_flutter_test.dart

```diff
@@ -115,20 +287,44 @@ class FakePlatformWebView {
 
   MethodChannel channel;
 
-  String lastUrlLoaded;
+  List<String> history = <String>[];
+  int currentPosition = -1;
+  String get lastUrlLoaded =>
+      (currentPosition > -1 && history.length > currentPosition)
```

Comment:
I'd just have this be:
```dart
String get currentUrl => history.isEmpty ? null : history[currentPosition];
```

We should be able to trust `currentPosition` to be valid here as we are the only ones maintaining it.